src/System.Time/src/System/Date.cs

```diff
@@ -0,0 +1,1629 @@
+        /// For example, if the current instance represents <c>2001-01-01</c>,
+        /// there has been one year since <c>2000-01-01</c>.
+        /// </remarks>
+        public int YearsSince(Date date)
+        {
+            return date.YearsUntil(this);
+        }
+
+        /// <summary>
+        /// Determines if a date falls within the range provided.
+        /// </summary>
+        /// <param name="startDate">The starting date, inclusive.</param>
+        /// <param name="endDate">
+        /// The ending date, either inclusive or exclusive, depending on the value of
+        /// <paramref name="exclusiveEndDate"/>.
+        /// </param>
+        /// <param name="exclusiveEndDate">
+        /// If true, then the <paramref name="endDate"/> is treated as exclusive.
+        /// Otherwise, the <paramref name="endDate"/> is treated as inclusive.
+        /// The default is <c>false</c> (inclusive).
+        /// </param>
+        /// <returns>True, if the date falls within the range, false otherwise.</returns>
+        /// <remarks>
+        /// Because a <see cref="Date"/> is intended to represent a whole calendar date,
+        /// this operation is inclusive by default.  For example, January 7th would be
+        /// considered to be included in the week of January 1st through January 7th.
+        /// If you desire the end date to be excluded, then set the <paramref name="exclusiveEndDate"/>
+        /// parameter to <c>true</c>.
+        /// </remarks>
+        public bool IsBetween(Date startDate, Date endDate, bool exclusiveEndDate = false)
+        {
+            return this >= startDate && (this < endDate || (this == endDate && !exclusiveEndDate));
+        }
+        
+        /// <summary>
+        /// Compares two instances of <see cref="Date"/> and returns an integer that indicates whether the first
+        /// instance is earlier than, the same as, or later than the second instance.
+        /// </summary>
+        /// <param name="d1">The first object to compare.</param>
+        /// <param name="d2">The second object to compare.</param>
+        /// <returns>
+        /// A signed number indicating the relative values of <paramref name="d1"/> and <paramref name="d2"/>.
+        /// <list type="table">
+        /// <listheader><term>Value</term><term>Description</term></listheader>
+        /// <item>
+        ///   <term>Less than zero</term>
+        ///   <term><paramref name="d1"/> is earlier than <paramref name="d2"/>.</term>
+        /// </item>
+        /// <item>
+        ///   <term>Zero</term>
+        ///   <term><paramref name="d1"/> is the same as <paramref name="d2"/>.</term>
+        /// </item>
+        /// <item>
+        ///   <term>Greater than zero</term>
+        ///   <term><paramref name="d1"/> is later than <paramref name="d2"/>.</term>
+        /// </item>
+        /// </list>
+        /// </returns>
+        public static int Compare(Date d1, Date d2)
```

Comment:
Acknowledged.  I was matching conventions of `System.DateTime` but I will update.  Should those be changed as well?